src/sentry/integrations/pipeline.py

```diff
@@ -88,12 +88,15 @@ def _finish_pipeline(self, data):
             }
 
             try:
-                Identity.objects.get_or_create(
+                ident, created = Identity.objects.get_or_create(
                     idp=idp,
                     user=self.request.user,
                     external_id=identity['external_id'],
                     defaults=identity_data,
                 )
+
+                if not created:
+                    ident.update(data=identity['data'], scopes=identity['scopes'])
```

Comment:
this whole thing can probably be more nicely expressed as
```
try:
    ident = Identity.get(...)
except DoesNotExist:
    relinking logic
else:
    ident.update(...)
```

Reply:
it's a little more than that, actually needs a case for both `DoesNotExist` (handle create) and `IntegrityError` (handle relinking)